src/otx/core/data/dataset/object_detection_3d.py

```diff
@@ -0,0 +1,307 @@
+        image = self._get_img_data_and_shape(image)[0]
+        calib = Calibration(entity.attributes["calib_path"])
+        original_kitti_format = None  # don't use for training
+        if self.subset_type != "train":
+            # TODO (Kirill): remove this or duplication of the inputs
+            annotations_copy = deepcopy(entity.annotations)
+            original_kitti_format = [obj.attributes for obj in annotations_copy]
+            # decode original kitti format for metric calculation
+            for i, anno_dict in enumerate(original_kitti_format):
+                anno_dict["name"] = self.label_info.label_names[annotations_copy[i].label]
+                anno_dict["bbox"] = annotations_copy[i].points
+                dimension = anno_dict["dimensions"]
+                anno_dict["dimensions"] = [dimension[2], dimension[0], dimension[1]]
+            original_kitti_format = self._reformate_for_kitti_metric(original_kitti_format)
+        # decode labels for training
+        inputs, targets, ori_img_shape = self._decode_item(
+            PILImage.fromarray(image),
+            entity.annotations,
+            calib,
+        )
+        # normilize image
+        inputs = self._apply_transforms(torch.as_tensor(inputs, dtype=torch.float32))
+        return Det3DDataEntity(
+            image=inputs,
+            img_info=ImageInfo(
+                img_idx=index,
+                img_shape=inputs.shape[1:],
+                ori_shape=ori_img_shape,  # TODO(Kirill): curently we use WxH here, make it HxW
+                image_color_channel=self.image_color_channel,
+                ignored_labels=[],
+            ),
+            boxes=tv_tensors.BoundingBoxes(
+                targets["boxes"],
+                format=tv_tensors.BoundingBoxFormat.XYXY,
+                canvas_size=inputs.shape[1:],
+                dtype=torch.float32,
+            ),
+            labels=torch.as_tensor(targets["labels"], dtype=torch.long),
+            calib_matrix=torch.as_tensor(calib.P2, dtype=torch.float32),
+            boxes_3d=torch.as_tensor(targets["boxes_3d"], dtype=torch.float32),
+            size_2d=torch.as_tensor(targets["size_2d"], dtype=torch.float32),
+            size_3d=torch.as_tensor(targets["size_3d"], dtype=torch.float32),
+            depth=torch.as_tensor(targets["depth"], dtype=torch.float32),
+            heading_angle=torch.as_tensor(
+                np.concatenate([targets["heading_bin"], targets["heading_res"]], axis=1),
+                dtype=torch.float32,
+            ),
+            original_kitti_format=original_kitti_format,
+        )
+
+    @property
+    def collate_fn(self) -> Callable:
+        """Collection function to collect DetDataEntity into DetBatchDataEntity in data loader."""
+        return partial(Det3DBatchDataEntity.collate_fn, stack_images=self.stack_images)
+
+    def _decode_item(self, img: PILImage, annotations: list[Bbox], calib: Calibration) -> tuple:  # noqa: C901
```

Comment:
I think some part of this function should be moved to data pipeline. How do you think about it?